security-advisories/README.md

```diff
@@ -24,6 +24,380 @@ Likewise you will find when it was fixed and who reported the issue.
+(shdr[sym_tab_idx].sh_addr + shdr[sym_tab_idx].sh_size) and
+(shdr[rel_sidx].sh_addr + shdr[rel_sidx].sh_size) could overflow, resulting in a
+large num_syms, or an invalid rel_end. Both could be used to access beyond
+legitimate memory. Outcome of such flaw is unclear but could be used to snoop or
+alter memory.
+
+**optee_os.git:**
+ - [core: ELF relocation: use ADD_OVERFLOW()](781c8f007c)(
+https://github.com/OP-TEE/optee_os/commit/781c8f007c4b4ac1d1966f1aacd37fbfe5d628aa)
+
+| Reported by                 | CVE ID    | OP-TEE ID        | Affected versions  |
+| --------------------------- | :-------: | :--------------: | ------------------ |
+| [Netflix] (Bastien Simondi) | Not/Ready | OP-TEE-2019-0015 | v3.5.0 and earlier |
+
+### ehdr.e_shnum could be very large and used to access out of bound memory
+At the end of elf_load_body(), code process relocation information
+segment: the relocation information are copied in a system heap memory block,
+associated to state->shdr. As the computed size is the result of an uncontrolled
+multiplication (ehdr.e_shnum * ehdr.e_shentsize), it could have overflowed and
+result in allocating a small memory block.
+
+Later in the code, there are no MUL_OVERFLOW() check either performed, and the
+code will access beyond the allocated memory. (e.g. in elf_process_rel() ) The
+outcome of this flaw depends on system heap memory arrangement but could be used
+to snoop information or alter memory.
+
+e.g. e32_process_rel() retrieved sym_tab_idx from shdr[rel_sidx].sh_link, then
+check if sym_tab_idx is smaller than ehdr->e_shnum; as the later can be as large
+as needed, shdr[sym_tab_idx] can point beyond the allocated memory block.
+Additionally, depending on memory arrangement, one could point to shared memory,
+and control the targeted memory block to mount further TOCTOU attacks (for
+instance to set sym_tab to an arbitrary value by updating
+shdr[sym_tab_idx].sh_addr with proper timing).
+
+**optee_os.git:**
+ - [core: elf_load_body(): use MUL_OVERFLOW() to get size of section headers](5787ecdf75)(
+https://github.com/OP-TEE/optee_os/commit/5787ecdf758d9edbfb5fb93c49c808f7a51a214b)
+
+| Reported by                 | CVE ID    | OP-TEE ID        | Affected versions  |
+| --------------------------- | :-------: | :--------------: | ------------------ |
+| [Netflix] (Bastien Simondi) | Not/Ready | OP-TEE-2019-0016 | v3.5.0 and earlier |
+
+### Integer overflow in TA memory map logic
+vm_map() and umap_add_region() do not check that given offs +
+ROUNDUP(len...) do not overflow. As a result the check to see if the region is
+in within a given memory object can be bypassed and both offset and/or size
+parameters could be very large.
+
+This could be leverage to alter the intended behavior of functions using either
+the region size or the region offset, like tee_mmu_user_pa2va_helper() for
+instance.
+
+**optee_os.git:**
+ - [core: umap_add_region(): add overflow check](bcc81cf8f0)(
```

Comment:
`]` closed too soon and missing space

Reply:
I've fixed all those. I was wondering why the link checker complained so much, now I know :)